From 571bd42e38e60b3e2a3f5a5a1cce9b0e21a9ad13 Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Wed, 3 Jul 2024 10:48:09 +0200
Subject: [PATCH] routing/cjdns: fix compiling on gcc8

Added condition to remove -Wno_overread warning if toolchain
is lower than version 11, based on
recommendation (https://github.com/NixOS/nixpkgs/pull/168997)
Added two patches from openwrt stream.
patch3.10 (1df5290b92625823bb9725a5406383116463f3ac)
invalid-pointer (d710a7938977c84d6ea0f74b21dc50111e5e968c)
In previous patch there was error with empty intcmp.
This commit fix error in in 26c38a17393e8771be6ed2cadb025c6c5e5f0b26
Rename TARGET_CCFLAGS to EXTRA_TARGET_CFLAGS, because TARGET_CCFLAGS is for C++ stuff.
---
 cjdns/Makefile                              | 11 ++++++++--
 cjdns/patches/030-fix-invalid-pointer.patch | 23 +++++++++++++++++++++
 cjdns/patches/040-gyp-python_310.patch      | 15 ++++++++++++++
 3 files changed, 47 insertions(+), 2 deletions(-)
 create mode 100644 cjdns/patches/030-fix-invalid-pointer.patch
 create mode 100644 cjdns/patches/040-gyp-python_310.patch

diff --git a/cjdns/Makefile b/cjdns/Makefile
index 1f8592b..6458a07 100644
--- a/cjdns/Makefile
+++ b/cjdns/Makefile
@@ -74,14 +74,21 @@ ifneq ($(CONFIG_USE_UCLIBC),)
 PKG_DO_VARS+= UCLIBC=1
 endif
 
+# Check if version compiler is ver. 8, then overread is ignored
+TOOLCHAIN_CC=$(word 1, $(subst ., ,${GCC_VERSION}))
+ifneq ($(TOOLCHAIN_CC), 8)
+NULL :=
+EXTRA_TARGET_CFLAGS=$(NULL) -Wno-error=stringop-overread
+endif
+
 define Build/Compile
 	$(INSTALL_DIR) $(PKG_BUILD_DIR)/tmp
 	(cd $(PKG_BUILD_DIR) && \
 	CROSS="true" \
-	CC="$(TARGET_CC)" \
+	CC="$(TARGET_CC_NOCACHE)" \
 	AR="$(TARGET_AR)" \
 	RANLIB="$(TARGET_RANLIB)" \
-	CFLAGS="$(TARGET_CFLAGS) -U_FORTIFY_SOURCE -Wno-error=array-bounds -Wno-error=stringop-overflow -Wno-error=stringop-overread" \
+	CFLAGS="$(TARGET_CFLAGS) -U_FORTIFY_SOURCE -Wno-error=array-bounds -Wno-error=stringop-overflow$(EXTRA_TARGET_CFLAGS)" \
 	LDFLAGS="$(TARGET_LDFLAGS)" \
 	SYSTEM="linux" \
 	TARGET_ARCH="$(CONFIG_ARCH)" \
-- 
2.45.2

